File: go-solution/SciPaperService/models/paper.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Paper struct {
	ID       primitive.ObjectID	`bson:"_id,omitempty" json:"id"`
	Author   string							`bson:"author" json:"author"`
	AuthorID string							`bson:"authorId" json:"authorId"`
	Title    string							`bson:"title" json:"title"`
	Sections []Section					`bson:"sections" json:"sections"`
}

type Section struct {
	Name    string	`bson:"name" json:"name"`
	Content string	`bson:"content" json:"content"`
}

File: dotnet-solution/ScipaperIO/SciPaperService/Models/Paper.cs
using System.Collections.Generic;

namespace SciPaperService.Models
{
    public class Paper : Document
    {
        public string Title { get; set; }
        public string Author { get; set; }
        public string AuthorId { get; set; }
        public List<Section> Sections { get; set; }
    }

    public class Section
    {
        public string Name { get; set; }
        public string Content { get; set; }
    }
}
